libfreerdp/core/capabilities.c

```diff
@@ -1294,23 +1294,16 @@ static BOOL rdp_read_input_capability_set(wStream* s, rdpSettings* settings)
 			settings->FastPathInput = FALSE;
 		}
 
+		/* Note: These settings have split functionality:
+		 * 1. If disabled in client pre_connect, it can disable announcing the feature
+		 * 2. If enabled in client pre_connect, override it with the server announced support flag.
+		 */
 		if (settings->HasHorizontalWheel)
-		{
-			if (inputFlags & TS_INPUT_FLAG_MOUSE_HWHEEL)
-				settings->HasHorizontalWheel = TRUE;
-		}
-
+			settings->HasHorizontalWheel = (inputFlags & TS_INPUT_FLAG_MOUSE_HWHEEL) ? TRUE : FALSE;
```

Comment:
`settings->HasHorizontalWheel = !!(inputFlags & TS_INPUT_FLAG_MOUSE_HWHEEL)` ?

Reply:
what do you mean?
only set this to `TRUE` if it was already requested in `preconnect`, otherwise let it stay `FALSE` and send that back?

Reply:
oh sorry I was not clear, I was just suggesting to have the boolean conversion done using `!!`